spec/2_system_overview.md

```diff
@@ -1,3 +1,51 @@
+
+### Oracle Scripts
+
+An oracle script is an executable program that encodes (1) the set of raw data requests to the sources it needs and (2) the way to aggregate raw data reports into the final result. These sources can be the data sources published on the network, as well as other oracle scripts. This composability makes oracle scripts very similar to [smart contracts](https://en.wikipedia.org/wiki/Smart_contract).
+
+### Oracle Data Request Initialization
+
+A data request to Band's oracle begins when an external party requests data from our oracle by broadcasting `MsgRequestData` *(TODO: LINK TO MSGREQUESTDATA DESCRIPTION)*. The contents of the message includes the oracle script the requester wants to invoke and other query and security parameters.
+
+Once the data transaction is confirmed on BandChain, the requested oracle script will begin its execution, which can be split into two phases. 
+
+### Fetching Data from Specified Data Sources
+
+First, the oracle script’s preparation function will emit the set of raw data requests necessary to continue the oracle script’s execution.The chain’s validators, who are chosen at random for security reasons, will then inspect the raw data requests and execute the associated data sources' procedures as instructed by the request. Each validator that successfully retrieved the data will then submit a raw data report to BandChain, containing the results they got from each of the sources, by broadcasting MsgReportData *(TODO: LINK TO MSGREPORTDATA DESCRIPTION)*. Once a sufficient number of validators, specified in the data request’s security parameters, have reported the raw data results, Once a sufficient number of reports are collected, this will resume BandChain’s execution of the oracle script and begin the script’s second phase of execution.
+
+### Saving the request result onto BandChain
+
+This second phase begins by aggregating all of the validators' reports into a final singular result. This final result is then permanently stored in BandChain’s application state. Once stored, the result becomes available on the BandChain’s state tree and can be sent to other blockchains through Inter-Blockchain Communication *(TODO: LINK TO IBC SECTION)*.
+
+![Oracle Data Request Flow](assets/oracle_request_flow.png)
+
+When the final result is successfully stored, an oracle data proof is also produced. This proof is a Merkle proof that shows the existence of the final result of the data request as well as other related information (oracle script hash, the parameters, the time of execution, etc) on BandChain. This proof can then be used by smart contracts on other blockchain to verify the existence of the data as well as to decode and retrieve the result stored. Both of these can be done by interacting with our lite client *(TODO: LINK TO LITE CLIENT SECTION)*.
```

Comment:
> This proof can then be used by smart contracts on other blockchain to verify the existence of the data as well as to decode and retrieve the result stored.

Smart contracts on other blockchains can use the proof to verify the validity of the data as well as to decode and retrieve the result stored.

Reply:
Use active voice